neo4j/src/main/java/org/hibernate/ogm/datastore/neo4j/remote/dialect/impl/RemoteNeo4jSequenceGenerator.java

```diff
@@ -0,0 +1,253 @@
+/*
+ * Hibernate OGM, Domain model persistence for NoSQL datastores
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.ogm.datastore.neo4j.remote.dialect.impl;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+import org.hibernate.HibernateException;
+import org.hibernate.boot.model.relational.Sequence;
+import org.hibernate.internal.util.collections.BoundedConcurrentHashMap;
+import org.hibernate.ogm.datastore.neo4j.dialect.impl.BaseSequenceGenerator;
+import org.hibernate.ogm.datastore.neo4j.dialect.impl.NodeLabel;
+import org.hibernate.ogm.datastore.neo4j.remote.impl.Neo4jClient;
+import org.hibernate.ogm.datastore.neo4j.remote.json.impl.Statement;
+import org.hibernate.ogm.datastore.neo4j.remote.json.impl.StatementResult;
+import org.hibernate.ogm.datastore.neo4j.remote.json.impl.Statements;
+import org.hibernate.ogm.datastore.neo4j.remote.json.impl.StatementsResponse;
+import org.hibernate.ogm.id.impl.OgmSequenceGenerator;
+import org.hibernate.ogm.id.impl.OgmTableGenerator;
+import org.hibernate.ogm.model.key.spi.IdSourceKey;
+import org.hibernate.ogm.model.key.spi.IdSourceKeyMetadata;
+import org.hibernate.ogm.model.key.spi.IdSourceKeyMetadata.IdSourceType;
+import org.neo4j.graphdb.DynamicLabel;
+import org.neo4j.graphdb.Label;
+
+/**
+ * Generates the next value of an id sequence as represented by {@link IdSourceKey}.
+ * <p>
+ * Both, {@link IdSourceType#TABLE} and {@link IdSourceType#SEQUENCE} are supported. For the table strategy, nodes in
+ * the following form are used (the exact property names and the label value can be configured using the options exposed
+ * by {@link OgmTableGenerator}):
+ *
+ * <pre>
+ * (:hibernate_sequences:TABLE_BASED_SEQUENCE { sequence_name = 'ExampleSequence', current_value : 3 })
+ * </pre>
+ *
+ * For the sequence strategy, nodes in the following form are used (the sequence name can be configured using the option
+ * exposed by {@link OgmSequenceGenerator}):
+ *
+ * <pre>
+ * (:SEQUENCE { sequence_name = 'ExampleSequence', next_val : 3 })
+ * </pre>
+ *
+ * Sequences are created at startup.
+ * <p>
+ * A write lock is acquired on the node every time the sequence needs to be updated.
+ *
+ * @author Davide D'Alto &lt;davide@hibernate.org&gt;
+ * @author Gunnar Morling
+ */
+public class RemoteNeo4jSequenceGenerator extends BaseSequenceGenerator {
+
+	/**
+	 * Query for creating SEQUENCE nodes.
+	 */
+	protected static final String SEQUENCE_CREATION_QUERY = "MERGE (n:" + NodeLabel.SEQUENCE.name() + " {" + SEQUENCE_NAME_PROPERTY + ": {sequenceName}} ) RETURN n";
```

Comment:
Why is the creation query different from the embedded one? It doesn't take into account initialValue?